lib/transform-stream.js

```diff
@@ -2,30 +2,62 @@
 /* global TransformStream */
 
 /**
- * Pipe a readable stream through a transformer. Return the readable end of the
- * TransformStream.
+ * Pipe a readable stream through a transformer. Returns a result, where
+ * result.readable is the readable end of the TransformStream and
+ * result.done is a promise that fulfills or rejects once the stream is done.
  * Includes a shim for environments where TransformStream is not available.
  */
-export function transformStream (readable, transformer) {
-  // Chrome, Edge, Safari TP
+export function transformStream (sourceReadable, transformer) {
+  let transformedReadable
+  let done
+
   if (typeof TransformStream !== 'undefined') {
-    return readable.pipeThrough(new TransformStream(transformer))
+    // Chrome, Edge, Safari 14.1+
+    const transform = new TransformStream(transformer)
+
+    done = sourceReadable.pipeTo(transform.writable)
+    transformedReadable = transform.readable
+  } else {
+    // Firefox, Safari 14 and older
+    let doneCb = null
+
+    done = new Promise((resolve, reject) => {
+      doneCb = (err) => {
+        if (err) {
+          reject(err)
+        } else {
+          resolve()
+        }
+      }
+    })
+    transformedReadable = new ReadableStream(new TransformStreamSource(sourceReadable, transformer, doneCb))
```

Comment:
Maybe cleaner to pass `{ resolve, reject }` into TransformStreamSource directly? But I'm not sure.

Reply:
I'm not sure, but I'll change it